typescript/sdk/src/providers/MultiProvider.ts

```diff
@@ -308,34 +334,55 @@ export class MultiProvider<MetaExt = {}> extends ChainMetadataManager<MetaExt> {
    * Wait for deploy tx to be confirmed
    * @throws if chain's metadata or signer has not been set or tx fails
    */
-  async handleDeploy<F extends ContractFactory>(
+  async handleDeploy<F extends ZKSyncContractFactory | ContractFactory>(
     chainNameOrId: ChainNameOrId,
     factory: F,
     params: Parameters<F['deploy']>,
+    artifact?: ZKSyncArtifact,
   ): Promise<Awaited<ReturnType<F['deploy']>>> {
-    // setup contract factory
+    const metadata = this.tryGetChainMetadata(chainNameOrId);
+    if (!metadata) {
+      throw new Error('Chain metadata not found!');
```

Comment:
Just use `this.getChainMetadata(chainNameOrId);` and it'll throw if if doesn't find it

Reply:
Resolved